### PR TITLE
feat: feature type updated audit log

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Added audit for feature type update

Short version:
Screenshot 2023-11-24 at 13 56 35

Full event:
Screenshot 2023-11-24 at 14 05 24

Design decisions:
* FEATURE_TYPE_UPDATED event added instead of more granular LIFETIME_UPDATED. We may have other changes soon e.g. name change.
* added getRecordedEvents helper method to test app to easily check if the correct event got recorded
* added extra query in a service to get the preData status of the event type
* I decided to add tx support and refactor towards feature oriented architecture in a separate PR

### CODE REVIEW

**Review by gastonfournier — APPROVED** [review]
LGTM!

File: src/lib/services/feature-type-service.ts [hunk 1]
```diff
@@ -27,23 +34,33 @@ export default class FeatureTypeService {
...
             id,
             translatedLifetime,
         );
 
-        if (!result) {
+        if (!featureType || !result) {
             throw new NotFoundError(
                 `The feature type you tried to update ("${id}") does not exist.`,
             );
         }
```
Comment (thomasheartman) [inline src/lib/services/feature-type-service.ts:55]: If this is the only error case here, would it make sense to move it to before trying to update the lifetime and only check whether it exists? 


```suggestion
        if (!featureType) {
            throw new NotFoundError(
                `The feature type you tried to update ("${id}") does not exist.`,
            );
        }
```

**Review by thomasheartman — APPROVED** [review]
Nice one! 🙌🏼

File: src/lib/services/feature-type-service.ts (hunk 1 above)
Comment (kwasniew) [reply src/lib/services/feature-type-service.ts:55]: tried that originally but TS led me towards my current solution. You can try for yourself. If you add your line and move it before update the return statement will not compile

**Comment by ivarconr** [comment]
nice!

**Pull request merged**